src/effects/Effect.h

```diff
@@ -304,12 +304,9 @@ class AUDACITY_DLL_API Effect /* not final */
    // UI
    //! This smart pointer tracks the lifetime of the dialog
    wxWeakRef<wxDialog> mHostUIDialog;
-   wxWindow       *mUIParent{};
-
-public:
-   wxWindow* GetUIParent() { return mUIParent; }
 
 private:
+   wxWindow       *mUIParent{};
```

Comment:
I want to eliminiate Effect::mUIParent.  This pull request does not complete that.  It does remove direct uses of it in subclasses, which instead shadow it.  Stateless subclasses store it in the Validator object instead of the effect, which is even better, eliminating some state.

To eliminaite it entirely requires rewriting of Effect::MessageBox and its many uses.  That is to be done later.